hermes/probe/error/error.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove, GitHub: evanSpendlove.
+//
+// Error implements the ProbeError used in Hermes for recording the
+// exit status of functions.
+
+// Package error defines the ProbeError used in Hermes for recording the
+// exit status of functions.
+package error
+
+import (
+	"fmt"
+
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+)
+
+// ProbeError is an error that includes an exit status for used within Hermes.
+type ProbeError struct {
+	Status metrics.ExitStatus
+	Err    error
+}
+
+// Wrap returns a new ProbeError containing the error passed
+// with additional context.and the status.
+// Arguments:
+//	- status: pass the exit status associated with this error.
+//	- msg: pass the additional context of the error as a string.
+//	- innner: pass the inner error to be wrapped.
+// Returns:
+//	- ProbeError: returns a new ProbeError object containing the args passed.
+func Wrap(status metrics.ExitStatus, msg string, inner error) ProbeError {
+	return &ProbeError{
+		Status: status,
+		Err:    fmt.Errorf("%s: %w", msg, inner),
```

Comment:
This is not wrapping, as it's transforming the underlying error, losing its type information.
Wrapping means we retain the original value of `inner`, so it can be unwrapped.
Wrapping is lossless.

Reply:
The %w verb appears to do magical wrapping with fmt.Errorf: https://blog.golang.org/go1.13-errors

Reply:
Ah okay, I misunderstood the concept then. I've renamed this to `New()` as I think that is more appropriate.